### ISSUE TITLE
feature: Annotate Service exposed via Gateway

### ISSUE THREAD

**Issue by tamalsaha** [envoyproxy/gateway#2656]
*Description*:
The Gateway api can become complex and hard to follow. As a user, it will be very helpful if the gw controller could annotate Services that are exposed via a Gateway with the Gateway namespace, name, port/listener name. Ideally, this should probably be part of the Service status but bar for that is probably insurmountable. Thanks!

https://kubernetes.slack.com/archives/CR0H13KGA/p1708327443009099

>Any extra documentation required to understand the issue.

**Comment by arkodg** [comment]
this is already done today https://github.com/envoyproxy/gateway/blob/cf46fbe776918ad19444e26d637ffcc79676ca23/internal/infrastructure/kubernetes/proxy/testdata/services/default.yaml#L8

using these labels
```
    gateway.envoyproxy.io/owning-gateway-name: <>
    gateway.envoyproxy.io/owning-gateway-namespace: <>
```

**Comment by tamalsaha** [comment]
Thanks for the quick response. 

1. What happens if the same service is exposed by multiple Gateway?
2. Can the port number on the gateway listener also added to this?

**Comment by arkodg** [comment]
1. isn't possible unless you are using `mergeGateways`, for that case the labels will change to `gateway.envoyproxy.io/owning-gatewayclass-name`
2. Any other dynamic info can be derived by retrieving the parent owning this resource, static info (labels/annotations) can always be added by the user using `EnvoyProxy` resource of `infrastructure` stanza within the `Gateway` resource

**Comment by tamalsaha** [comment]
Hi @arkodg , I might have not explained it correctly. I am not talking about the GW proxy service. I am talking about annotating the backend services that are exposed via a gateway. Could the controller put some annotation on the backend service to indicate that it is exposed via a Gateway?

**Comment by zirain** [comment]
you want add some annotations to a service after it's expose by GW?

**Comment by tamalsaha** [comment]
> you want add some annotations to a service after it's expose by GW?

Yes

**Comment by zirain** [comment]
it's not 1:1, what would you expect when one service referenced by multi gateways?

**Comment by tamalsaha** [comment]
I would like all of them to the reflected in the annotation. It is security issue, if a service is exposed unintentionally / by mistake.
